### PR TITLE
feat: Add adjustable timeout into go client

### PR DESCRIPTION
**Pull request by Matovidlo**
**Changes:**
- Reduce linter concurrency, enable to log all linter issues.
- Use new go client with adjustable timeout for API calls.
- Add adjustable timeout in tests.
- Add timeout for `newProjectScope`.

-----------

### CODE REVIEW

File: internal/pkg/service/common/dependencies/public.go [hunk 1]
```diff
@@ -76,23 +76,23 @@ func newPublicScope(ctx context.Context, baseScp BaseScope, storageAPIHost strin
 	var indexWithComponents *keboola.IndexComponents
 	if cfg.preloadComponents {
 		logger.Info(ctx, "loading Storage API index with components")
-		indexWithComponents, err = keboola.APIIndexWithComponents(ctx, storageAPIHost, keboola.WithClient(&baseHTTPClient))
+		indexWithComponents, err = keboola.APIIndexWithComponents(ctx, storageAPIHost, keboola.WithClient(&baseHTTPClient), keboola.WithOnSuccessTimeout(1*time.Minute))
```
Comment (Matovidlo) [inline internal/pkg/service/common/dependencies/public.go:79]: How long should be timeout for API related `dependencies`?

File: scripts/lint.sh [hunk 2]
```diff
@@ -31,7 +31,7 @@ echo
 
 # Run linters
 echo "Running golangci-lint ..."
-if golangci-lint run -c "./build/ci/golangci.yml" "$@"; then
+if golangci-lint run -j 8 --max-issues-per-linter 0  --max-same-issues 0 -c "./build/ci/golangci.yml" "$@"; then
     echo "Ok. The code looks good."
```
Comment (michaljurecko) [inline scripts/lint.sh:35]: Use config file please.

File: internal/pkg/service/common/dependencies/public.go (hunk 1 above)
Comment (michaljurecko) [reply internal/pkg/service/common/dependencies/public.go:79]: The response may have up to `1-2` MBs, 30s?

File: go.mod [hunk 3]
```diff
@@ -40,7 +40,7 @@ require (
 	github.com/joho/godotenv v1.5.1
 	github.com/jpillora/longestcommon v0.0.0-20161227235612-adb9d91ee629
 	github.com/json-iterator/go v1.1.12
-	github.com/keboola/go-client v1.26.7
+	github.com/keboola/go-client v1.26.7-0.20240830125509-01ac1dc03025
```
Comment (Matovidlo) [inline go.mod:43]: https://github.com/keboola/go-client/pull/138

**Review by hosekpeter — APPROVED** [review]

**Pull request merged**